src/distribute_kv_cache/kvclient.rs

```diff
@@ -0,0 +1,970 @@
+            }
+
+            debug!("Failed to get block from local cache, try to get from the distribute cache");
+        }
+
+        // Empty address from the distribute cache
+        if node_address.is_empty() {
+            error!("Failed to get block, node address is empty");
+            return Ok((Vec::new(), bytes::Bytes::new()));
+        }
+
+        // 3. Get the block from the distribute cache
+        // TODO: update string key with u64
+        let block_data = match self
+            .inner
+            .get_block(node_address.clone(), kv_block_meta.block_id)
+            .await
+        {
+            Ok(data) => data,
+            Err(err) => {
+                // Inform master to delete this key
+                self.inner.remove_index(prefix).await?;
+
+                error!(
+                    "Failed to get block: {:?} node_address: {:?} kv_block_meta: {:?}",
+                    err, node_address, kv_block_meta
+                );
+                return Ok((Vec::new(), bytes::Bytes::new()));
+            }
+        };
+
+        // 4. Copy the block data to the data with kv cache meta offset and size
+        let offset = u64_to_usize(kv_block_meta.offset);
+        let size = u64_to_usize(kv_block_meta.size);
+        debug!(
+            "Get block from remote cache: {:?} offset: {:?} size: {:?}",
+            kv_block_meta, offset, size
+        );
+
+        Ok((
+            kv_block_meta.prefix,
+            block_data.slice(offset..offset + size),
+        ))
+    }
+
+    /// Insert a block to the distribute cache
+    #[allow(clippy::too_many_lines)]
+    pub async fn insert(&self, prefix: Vec<K>, data: Vec<u8>) -> DatenLordResult<()> {
+        // Check current size is valid
+        if data.len().cast::<u64>() > self.inner.block_size {
+            return Err(DatenLordError::DistributeCacheManagerErr {
+                context: vec![format!(
+                    "Failed to insert kv cache to the block cache, data size is too large"
+                )],
+            });
+        }
+
+        // 1. fill current block cache
+        let mut block_cache = self.block_cache.lock().await;
+        let mut current_block_id = block_cache.get_block_id();
+        debug!(
+            "Insert kv cache to the block cache with block id: {:?}",
+            current_block_id
+        );
+
+        while current_block_id == UNUSED_KV_BLOCK_ID {
+            let new_block_id = self.inner.alloc_block_id().await?;
+            debug!("Alloc new block id: {:?}", new_block_id);
+            block_cache.clear(new_block_id);
+            current_block_id = new_block_id;
+        }
+
+        let next_offset = block_cache.get_next_offset();
+        let kv_cache_meta = KVCacheMeta {
+            block_id: current_block_id,
+            offset: next_offset,
+            size: data.len().cast(),
+            prefix: prefix.clone(),
+        };
+
+        match block_cache.insert(kv_cache_meta, &data) {
```

Comment:
block_cache's `clear` and `insert` always called together, why don't you merge them?

Reply:
I prefer merging these two functions.

Reply:
OK, changed to `clear_and_insert`.

Reply:
After confirmation, the clear should still be separated here. Only when insert encounters an error (after being filled) will it clear and reapply for a block ID. If there is still remaining space, there is no need to clear the block.

Reply:
Change `clear()` to `refresh_id_and_clear()`.